### PR TITLE
provisioner/shell: remove file after exec [GH-1536]

### LINKED ISSUE #1536: Shell provisioner leaves behind temporary script
The shell provisioner creates a script /tmp/script.sh and then executes it.  It would be an improvement if the script was deleted after executing so that subsequent uses of the shell provisioner, possibly by another user, does not step on it.  I see this warning in the logs:

chmod: changing permissions of `/tmp/script.sh': Operation not permitted

This happens when I initially create an image using qemu and root login and then build on top of that image using a non root user.

### PR DESCRIPTION
**Pull request by mitchellh**
Fixes #1536 

This deletes the temporary shell script after running.

### CODE REVIEW

File: provisioner/shell/provisioner.go [hunk 1]
```diff
@@ -266,12 +266,24 @@ func (p *Provisioner) Provision(ui packer.Ui, comm packer.Communicator) error {
...
+		}
+		if err := comm.Start(cmd); err != nil {
+			return fmt.Errorf(
+				"Error removing temporary script at %s: %s",
+				p.config.RemotePath, err)
+		}
+		cmd.Wait()
+		if cmd.ExitStatus != 0 {
+			return fmt.Errorf(
+				"Error removing temporary script at %s!")
```
Comment (cbednarski) [inline provisioner/shell/provisioner.go]: There's no argument for `%s` here.

Comment (mitchellh) [inline provisioner/shell/provisioner.go]: Fixed

**Comment by cbednarski** [comment]
LGTM

**Comment by roumano** [comment]
This new function break the possibility to "delete the user" used to connect on the builders

For exemple, on official amazon AMI, we need to connect with these user 
CentOS7 == centos
Debian == admin
OEL6 , RHEL6 , RHEL7 == ec2-user

Our last provisioners module (on these OS) is to delete this user
Now we can't build as we can't connect to delete it

Can you add a possibility to not remove file after exec 
Or even better a possibility to not stop the building process as it's not a important issue if a file is still on /temp.

Regards

**Pull request merged**